tests/crypto_policy_test.go

```diff
@@ -0,0 +1,238 @@
+package tests
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	ocpv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/crypto"
+	core "k8s.io/api/core/v1"
+	apiregv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
+	sspv1beta1 "kubevirt.io/ssp-operator/api/v1beta1"
+	"kubevirt.io/ssp-operator/internal/common"
+	"kubevirt.io/ssp-operator/tests/env"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+var _ = Describe("Crypto Policy", func() {
+	BeforeEach(func() {
+		waitUntilDeployed()
+	})
+
+	FContext("setting Crypto Policy", func() {
+		sspPod := operatorPod()
+		table.DescribeTable("Adhere to defined TLSConfig", testSettingTLSConfig,
+			table.Entry("[test_id:???] old", sspPod, tlsConfigTestPermutations["Old"]),
+			table.Entry("[test_id:???] intermediate", sspPod, tlsConfigTestPermutations["Intermediate"]),
+			table.Entry("[test_id:???] modern", sspPod, tlsConfigTestPermutations["Modern"]),
+		)
+	})
+})
+
+func operatorPod() core.Pod {
+	cfg, err := config.GetConfig()
+	Expect(err).ToNot(HaveOccurred())
+	apiClient, err = client.New(cfg, client.Options{Scheme: testScheme})
+	Expect(err).ToNot(HaveOccurred())
+	pods := &core.PodList{}
+	err = apiClient.List(context.TODO(), pods, client.MatchingLabels{"control-plane": "ssp-operator"})
+	Expect(err).ToNot(HaveOccurred())
+	Expect(pods.Items).ToNot(BeEmpty())
+	return pods.Items[0]
+}
+
+type tlsConfigTestPermutation struct {
+	openshiftTLSPolicy                *ocpv1.TLSSecurityProfile
+	allowedConfigs, disallowedConfigs []clientTLSOptions
+}
+
+type clientTLSOptions struct {
+	MaxTLSVersion      string
+	OpenSSLCipherNames []string
+}
+
+func (s *clientTLSOptions) MaxTLSVersionId() (uint16, error) {
+	switch s.MaxTLSVersion {
+	case "":
+		return tls.VersionTLS10, nil
+	case "1.0":
+		return tls.VersionTLS10, nil
+	case "1.1":
+		return tls.VersionTLS11, nil
+	case "1.2":
+		return tls.VersionTLS12, nil
+	case "1.3":
+		return tls.VersionTLS13, nil
+	default:
+		return 0, fmt.Errorf("invalid TLSMinVersion %v: expects 1.0, 1.1, 1.2, 1.3 or empty", s.MaxTLSVersion)
+	}
+}
+
+func (s *clientTLSOptions) CipherIDs(logger *logr.Logger) (cipherSuites []uint16) {
```

Comment:
I know this is a direct copy from the code above but could you remove the logger? It is never used in this test.